pkg/extensions/tbtc/recovery/storage.go

```diff
@@ -0,0 +1,106 @@
+package recovery
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+const (
+	fileName = "derivation_indexes"
+)
+
+type DerivationIndexStorage struct {
+	path string
+}
+
+func NewDerivationIndexStorage(path string) (*DerivationIndexStorage, error) {
+	err := ensureDirectoryExists(fmt.Sprintf("%s/%s", path, fileName))
+	if err != nil {
+		return nil, err
+	}
+	return &DerivationIndexStorage{
+		path: path,
+	}, nil
+}
+
+func (dis DerivationIndexStorage) getStoragePath(extendedPublicKey string) string {
+	publicKeyDescriptor := extendedPublicKey[:4]
+	suffix := extendedPublicKey[len(extendedPublicKey)-8:]
```

Comment:
Would be good to document how the key is transformed into directory name on some example.